multiuser/machine-auth/che-multiuser-machine-authentication/src/main/java/org/eclipse/che/multiuser/machine/authentication/server/signature/SignatureAlgorithmEnvProvider.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright (c) 2012-2018 Red Hat, Inc.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.che.multiuser.machine.authentication.server.signature;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import org.eclipse.che.api.core.model.workspace.runtime.RuntimeIdentity;
+import org.eclipse.che.api.workspace.server.spi.provision.env.EnvVarProvider;
+import org.eclipse.che.commons.lang.Pair;
+
+/**
+ * Provides signature algorithm into machine environment.
+ *
+ * @author Anton Korneta
+ */
+public class SignatureAlgorithmEnvProvider implements EnvVarProvider {
+
+  private static final String SIGNATURE_ALGORITHM_ENV = "SIGNATURE_ALGORITHM";
```

Comment:
I think it's better to name env variable in Che env var naming style, like `CHE_MACHINE_AUTH_SIGNATURE__ALGORITHM.
Or something simpler but Che related `CHE_SIGNATURE_ALGORITHM`